### DISCUSSION TITLE
PostCSS config file format limits plugin configuration and availability

### DISCUSSION THREAD

**Discussion by argyleink** [vercel/next.js#32585] [Help] [answered]
If a PostCSS plugin has dependencies on an NPM package, I don't see a way to configure that within the formats allowed in `postcss.config.js`. 

I'm trying to get the following PostCSS setup working inside of Next: [stackblitz](https://stackblitz.com/edit/jit-open-props?file=postcss.config.js)

```js
const postcssJitProps = require('postcss-jit-props');
const OpenProps = require('open-props');

module.exports = {
  plugins: [postcssJitProps(OpenProps, { custom_selector: ':global' })],
};
```

Notice how the plugin needs passed the npm package plus a 2nd object for configuration options. 

I keep trying various array and object formats to get this to work inside Next, and havent found the right combination yet or have reached the edge of the capabilities of Nextjs and PostCSS configuration?

**Comment by pcjmfranken** [comment, marked as answer]
Noticed you fixed it yourself in this particular case https://github.com/GoogleChromeLabs/postcss-jit-props/pull/8. Interesting plugins by the way, hadn't seen them before!

PostCSS doesn't pose any limits on plugin input parameters so the plugin's previous implementation was unconventional but not wrong. For any other incompatible plugins that can't or are unwilling to accept such a change, you could always write your own wrapper plugin[^1] that takes a single param and redistributes it in the required format.

The plugin config format that NextJS requires - the only variant with this limitation - seems to be deprecated [^2], so it's going to have to change when Webpack 6 is out anyway.

[^1]: PostCSS Plugin docs: https://github.com/postcss/postcss/blob/main/docs/writing-a-plugin.md#step-2-create-a-project
[^2]: Deprecation warning can be seen here: https://webpack.js.org/loaders/postcss-loader/#object
